main.py

```diff
@@ -85,7 +86,7 @@ def execute(self):
         log.debug("Finished consistency routine")
 
     @staticmethod
-    def should_be_checked(circuit):
+    def should_be_checked(circuit, switches):
```

Comment:
Perhaps, rather than having `should_be_checked` as a static method, instead making it an instance method, so it no longer needs to have switches as an argument, and instead can retrieve it from self.

Reply:
Got it, changed to non-static method.